pkg/event-manager/drivers/entity_handler.go

```diff
@@ -100,13 +106,142 @@ func (h *EntityHandler) Delete(obj entitystore.Entity) error {
 func (h *EntityHandler) Sync(organizationID string, resyncPeriod time.Duration) ([]entitystore.Entity, error) {
 	defer trace.Trace("")()
 
-	return controller.DefaultSync(h.store, h.Type(), organizationID, resyncPeriod, nil)
+	// list entity filter
+	filter := entitystore.FilterEverything().Add(
+		entitystore.FilterStat{
+			Scope:   entitystore.FilterScopeField,
+			Subject: "ModifiedTime",
+			Verb:    entitystore.FilterVerbBefore,
+			Object:  time.Now().Add(-resyncPeriod),
+		},
+		entitystore.FilterStat{
+			Scope:   entitystore.FilterScopeField,
+			Subject: "Status",
+			Verb:    entitystore.FilterVerbIn,
+			Object: []entitystore.Status{
+				entitystore.StatusCREATING, entitystore.StatusUPDATING, entitystore.StatusDELETING,
+				entitystore.StatusERROR,
+				entitystore.StatusREADY,
+			},
+		})
+	syncingEntities, err := controller.DefaultSync(h.store, h.Type(), organizationID, resyncPeriod, filter)
+	if err != nil {
+		return nil, err
+	}
+
+	// sync with k8s, pending entities might change status during sync
+	k := h.backend.(*k8sBackend)
```

Comment:
Let's avoid leaking the implementation of backend. What if we wanted to use different (non-k8s) backend here? I think we could use `Update()` method of `Backend` interface to implement the logic of reconciliation (we could extend that interface as well)